uncover/agent/censys/censys.go

```diff
@@ -75,7 +69,10 @@ func (agent *Agent) queryURL(session *uncover.Session, URL string, censysRequest
 	}
 	request.Header.Set("Accept", "application/json")
 	request.SetBasicAuth(session.Keys.CensysToken, session.Keys.CensysSecret)
-	agent.options.RateLimiter.Take()
+	err = session.RateLimits.Take(agent.Name())
+	if err != nil {
+		return nil, err
+	}
 	return session.Do(request)
 }
```

Comment:
same here , I think we can modify session.Do to accept a optional sourcename and move ratelimit logic inside `Do` function

```go
func (s *Session) Do(req *retryablehttp.Request,source ...string)
```